File: BACK/backend/Proyecto-Gestion-Ambiental/databasefirst/Models/Participacion.cs
using System;
using System.Collections.Generic;

namespace databasefirst.Models;

public partial class Participacion
{
    public int id_participacion { get; set; }

    public int? id_user { get; set; }

    public int? id_actividad { get; set; }

    public DateTime? fecha_inscripcion { get; set; }

    public string? rol_en_actividad { get; set; }

    public string? observaciones { get; set; }

    public string? estado { get; set; }

    public virtual Actividad? id_actividadNavigation { get; set; }

    public virtual Usuario? id_userNavigation { get; set; }
}

File: FRONT/frontend/proyecto-Gestion-Ambiental/src/app/models/Participacion.ts
/**Autor:Gabriela Gonzalez
  * Fecha: 2026/01/27
  * Descripción: Modelo para representar participación en proyectos.
  */

export interface Participacion {
  id_participacion?: number; 
  id_user: number;
  id_actividad: number;
  fecha_inscripcion: string; 
  rol_en_actividad: string;
  observaciones?: string;   
  estado: string;             
  
  id_userNavigation?: {
    nombre: string;
    apellido: string;
    correo: string;
  };

  id_actividadNavigation?: {
    nombre: string;
    descripcion: string;
    lugar: string;
  };
}
